R/fa_i.R

```diff
@@ -24,19 +41,54 @@
 #' @export
 fa_i <- function(name,
                  class = NULL,
+                 htmlDependency = NULL,
                  ...) {
 
   prefix <- "fa"
   iconClass <- ""
 
-  if (!is.null(name)) {
-    prefix_class <- prefix
-    if (prefix_class == "fa" && name %in% font_awesome_brands) {
-      prefix_class <- "fab"
-    }
-    iconClass <- paste0(prefix_class, " ", prefix, "-", name)
+  # Determine if the `name` is a Font Awesome v4
+  # icon name and provide a warning
+  if (name %in% fa_tbl$v4_name && !(name %in% fa_tbl$name)) {
+
+    # Obtain the version 5 `name` and `full_name`
+    # for messaging purposes
+    v5_name <- fa_tbl[fa_tbl$v4_name == name, ][1, "name"]
```

Comment:
If the user is providing their own html dependency object, I think we shouldn't do these checks at all.

Reply:
Now changed, as discussed in person.